### PR TITLE
fix test resnet50 multi-threading on mkldnn

### PR DESCRIPTION
**Pull request by sfraczek**
I have also added printing FPS next to latency for convenience if you don't mind.

### CODE REVIEW

File: paddle/fluid/operators/parallel_do_op.cc [hunk 1]
```diff
@@ -18,6 +18,7 @@ limitations under the License. */
 #include "paddle/fluid/framework/op_registry.h"
```
Comment (luotao1) [inline paddle/fluid/operators/parallel_do_op.cc]: `parallel_do_op` is deprecated now, you could not update it.

File: paddle/fluid/platform/device_context.h [hunk 2]
```diff
@@ -39,6 +39,12 @@ limitations under the License. */
 namespace paddle {
 namespace platform {
 
+using KeyBlob = std::unordered_map<std::string, std::shared_ptr<void>>;
+using BlobMap = std::unordered_map<int, std::shared_ptr<KeyBlob>>;
+
+void set_cur_thread_id(int);
+int get_cur_thread_id(void);
```
Comment (luotao1) [inline paddle/fluid/platform/device_context.h]: line 42-46 are only used in `PADDLE_WITH_MKLDNN`. Could these line move after line 184?

File: paddle/fluid/inference/tests/api/tester_helper.h [hunk 3]
```diff
@@ -139,6 +139,7 @@ void TestMultiThreadPrediction(
   }
   for (int tid = 0; tid < num_threads; ++tid) {
     threads.emplace_back([&, tid]() {
+      platform::set_cur_thread_id(static_cast<int>(tid) + 1);
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/tester_helper.h:143]:
```
#ifdef PADDLE_WITH_MKLDNN
platform::set_cur_thread_id(static_cast<int>(tid) + 1);
#endif
```

File: paddle/fluid/platform/device_context.cc [hunk 4]
```diff
@@ -25,6 +25,14 @@ namespace platform {
 
 DeviceContextPool* DeviceContextPool::pool = nullptr;
 
+namespace {
+// Current thread's id.
+thread_local int cur_thread_id = 0;
+}
+
+void set_cur_thread_id(int tid) { cur_thread_id = tid; }
+int get_cur_thread_id(void) { return cur_thread_id; }
```
Comment (luotao1) [inline paddle/fluid/platform/device_context.cc]: line 30-34 are only used in PADDLE_WITH_MKLDNN. Could these line move after line 305?

**Review by luotao1 — COMMENTED** [review]
Do you have multi-instance performance data on ResNet50?

**Comment by bingyanghuang** [comment]
@luotao1 This is the benchmark I tested with this PR this morning on Xeon 2650 v4.
image

File: paddle/fluid/operators/parallel_do_op.cc (hunk 1 above)
Comment (sfraczek) [reply paddle/fluid/operators/parallel_do_op.cc]: ok I will remove this update

File: paddle/fluid/platform/device_context.h (hunk 2 above)
Comment (sfraczek) [reply paddle/fluid/platform/device_context.h]: ok

File: paddle/fluid/inference/tests/api/tester_helper.h (hunk 3 above)
Comment (sfraczek) [reply paddle/fluid/inference/tests/api/tester_helper.h:143]: ok

File: paddle/fluid/platform/device_context.cc (hunk 4 above)
Comment (sfraczek) [reply paddle/fluid/platform/device_context.cc]: ok

**Comment by luotao1** [comment]
please add `test=develop` in commit to trigger the CI

**Comment by luotao1** [comment]
I test on E5-2620 V3, the MKLDNN latency increase 20% on 8 threads, but increase 60% on 12 threads.
image
Do you know the reason?

**Comment by sfraczek** [comment]
This PR failed twice at 
```
[18:12:30][Step 1/1] AssertionError: importlib-resources; python_version < "3.7" .dist-info directory not found
[18:12:31][Step 1/1] The command '/bin/sh -c pip3 install pre-commit 'ipython==5.3.0' &&     pip3 install 'ipykernel==4.6.0' 'jupyter==1.0.0' &&     pip3 install opencv-python &&     pip install pre-commit 'ipython==5.3.0' &&     pip install 'ipykernel==4.6.0' 'jupyter==1.0.0' &&     pip install opencv-python' returned a non-zero code: 2
[18:12:31][Step 1/1] 
[18:12:31][Step 1/1] Process exited with code 2
[18:12:31][Step 1/1] Process exited with code 2
[18:12:31][Step 1/1] Step Build and test (Command Line) failed
```
@luotao1 is it a problem with CI? should I wait until it gets fixed?

**Comment by luotao1** [comment]
This machine is updated now, you can rerun the PR_CI.

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**